### PR TITLE
Add job ID to query struct

### PR DESCRIPTION
**Pull request by mahmednabil109**
jobID must be added to query in order to be used in filtering entries in the database

Signed-off-by: Mohamed Abokammer <mahmednabil109@gmail.com>

### CODE REVIEW

File: tests/integ/admin_server/getlogs_test.go [hunk 1]
```diff
@@ -100,6 +100,9 @@ func queryLogsEndpoint(query server.Query, retrieveAll bool) (*storage.Result, i
 		var tmp storage.Result
 
 		q := u.Query()
+		if query.JobID != nil {
+			q.Set("jobId", fmt.Sprint(*query.JobID))
```
Comment (mimir-d) [inline tests/integ/admin_server/getlogs_test.go:104]: nit; this works, but for clarity/future safety, let's do strconv.FormatUint

File: cmds/admin_server/server/server.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ var (
 )
 
 type Query struct {
+	JobID     *uint64    `form:"jobId"`
```
Comment (mimir-d) [inline cmds/admin_server/server/server.go:23]: didnt see this before; can we have snake_case for the rest api query? no reason in particular, it's a bit closer to json

**Review by mimir-d — CHANGES_REQUESTED** [review]

File: tests/integ/admin_server/logendpoint_test.go [hunk 3]
```diff
@@ -46,7 +47,7 @@ func setupCleanDB(uri string) (*mongo.Client, error) {
 	return client, err
 }
 
-func submitLog(addr string, log storage.Log) error {
+func submitLog(addr string, log server.Log) error {
```
Comment (mimir-d) [inline tests/integ/admin_server/logendpoint_test.go:50]: for this one, we should have it in a different PR. on this same topic, i see there's a storage.Log and a server.Log; the server one has json annotations, but are those needed on the storage model? if not, remove them in the same PR with this line

**Review by mimir-d — CHANGES_REQUESTED** [review]

File: tests/integ/admin_server/logendpoint_test.go (hunk 3 above)
Comment (mahmednabil109) [reply tests/integ/admin_server/logendpoint_test.go:50]: the json annotation is needed for serialization in both storage, and http response.

Comment (mimir-d) [reply tests/integ/admin_server/logendpoint_test.go:50]: done in #111

**Comment by mimir-d** [comment]
pease rebase on top of #111 (or main)

File: tests/integ/admin_server/logendpoint_test.go (hunk 3 above)
Comment (mahmednabil109) [reply tests/integ/admin_server/logendpoint_test.go]: rebased, but ig this will wait for #117

**Review by mimir-d — APPROVED** [review]

**Pull request merged**